flutter/test/mocks.dart

```diff
@@ -168,4 +171,9 @@ class MockNativeChannel implements SentryNativeChannel {
     numberOfEndNativeFramesCalls += 1;
     return nativeFrames;
   }
+
+  @override
+  FutureOr<void> setUser(SentryUser? user) {
```

Comment:
Won't this run synchronously? That is, until a call to something else that returns a Future is done?
That way,  this won't be a breaking change in terms of behavior. The Dart scope will be set synchronously, and once we add the call to the bridge code (async), that would require await if the code wants to continue only after the data has been sent down

Reply:
Yes, but this is the testing code only.
Right now the scope methods return `void`, it's not changed yet, so the user does not have the chance to `await`.
Once we change it to a `Future`, we'd need to introduce setters and this is the breaking change.
You're right, the behavior is the same since the scope methods will set the properties synchronously.
So, are fine adding setters and cutting a major change? do we fit more issues into this major?

https://github.com/getsentry/sentry-dart/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+milestone%3A7.0.0

We can probably move some of them from 7.x to 8.x since not all of them are really a blocker.

Reply:
@denrase We agreed that changing the return type is fine since the method is still called synchronously and the value is set to the scope, only the call to the method channels are fire and forget, unless the user `await` for the call, the call to method channels should be documented as best-effort if not `awaited`.
This should make it into the migration docs tho https://docs.sentry.io/platforms/flutter/migration/